src/utils/DatabaseFilter.cpp

```diff
@@ -81,15 +81,15 @@ class TOPPDatabaseFilter :
   {
     registerInputFile_("in", "<file>", "","Input FASTA file, containing a database.");
     setValidFormats_("in", ListUtils::create<String>("fasta"));
-    registerInputFile_("accession", "<file>", "", "Input IdXML file, containing the identfied peptides.", true);
-    setValidFormats_("accession", ListUtils::create<String>("idXML,mzid"));
-    registerStringOption_("method", "<type>", "whitelist", "Switch between white/blacklisting", false);
+    registerInputFile_("id", "<file>", "", "Input file containing identified peptides and proteins.");
+    setValidFormats_("id", ListUtils::create<String>("idXML,mzid"));
+    registerStringOption_("method", "<choice>", "whitelist", "Switch between white-/blacklisting of protein IDs", false);
     setValidStrings_("method", ListUtils::create<String>("whitelist,blacklist"));
     registerOutputFile_("out", "<file>", "", "Output FASTA file where the reduced database will be written to.");
     setValidFormats_("out", ListUtils::create<String>("fasta"));
   }
 
-  void filterByProteinAccessions_(const vector<FASTAFile::FASTAEntry>& db, const vector<PeptideIdentification>& peptide_identifications, bool whitelist, vector<FASTAFile::FASTAEntry>& db_new)
+  void filterByProteinIDs_(const vector<FASTAFile::FASTAEntry>& db, const vector<PeptideIdentification>& peptide_identifications, bool whitelist, vector<FASTAFile::FASTAEntry>& db_new)
```

Comment:
if you want to emphasize that IDs are used then ist should read filterByPeptideIDs_